File: shop_app/lib/entitys/product_entity.dart


class ProductEntity {
  int? id;
  int? price;
  String? date;
  int? amount;
  String? name;
  String? category;
  String? barCode;
  int? sameProduct;
  int? idShop;

  ProductEntity(
      {this.id, this.price, this.date, this.amount, this.name, this.category, this.barCode, this.sameProduct, this.idShop});

  ProductEntity.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    price = json['price'];
    date = json['date'];
    amount = json['amount'];
    name = json['name'];
    category = json['category'];
    barCode = json['barCode'];
    sameProduct = json['sameProduct'];
    idShop = json['idShop'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['price'] = this.price;
    data['date'] = this.date;
    data['amount'] = this.amount;
    data['name'] = this.name;
    data['category'] = this.category;
    data['barCode'] = this.barCode;
    data['sameProduct'] = this.sameProduct;
    data['idShop'] = this.idShop;
    return data;
  }
}

File: tienda-backend-apirest/src/main/java/com/bolsadeideas/springboot/backend/apirest/models/entity/ProductEntity.java
package com.bolsadeideas.springboot.backend.apirest.models.entity;

import javax.persistence.*;
import java.io.Serializable;


@Entity
@Table(name = "product")
public class ProductEntity  implements Serializable  {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long idShop;
    private Long price;
    private String date;
    private Long amount;
    private String name;
    private String category;
    private String barCode;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getIdShop() {
        return idShop;
    }

    public void setIdShop(Long idShop) {
        this.idShop = idShop;
    }

    public Long getPrice() {
        return price;
    }

    public void setPrice(Long price) {
        this.price = price;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public Long getAmount() {
        return amount;
    }

    public void setAmount(Long amount) {
        this.amount = amount;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getBarCode() {
        return barCode;
    }

    public void setBarCode(String barCode) {
        this.barCode = barCode;
    }

    @Override
    public String toString() {
        return "ProductEntity{" +
                "id=" + id +
                ", price=" + price +
                ", date='" + date + '\'' +
                ", amount=" + amount +
                ", name='" + name + '\'' +
                '}';
    }
}
